File: MultiLanguageSandbox/src/test/deps/javascript/6.js

/**
 * Find the factorial of N and take the modulo 10007 of the result.
 * 
 * Parameters:
 * - N (number): A non-negative integer representing the input value (N <= 10000).
 * 
 * Returns:
 * - number: The result after taking the modulo 10007 of the factorial of N.
 * 
 * Examples:
 * processRequest(1) returns 1
 */

function processRequest(n) {
    if (n < 0) {
        throw new Error("N must be a non-negative integer");
    }
    
    let result = 1;
    for (let i = 2; i <= n; i++) {
        result = (result * i) % 10007;
    }
    return result;
}
const testProcessRequest = () => {
    console.assert(processRequest(0) === 1, "Expected factorial of 0 modulo 10007 is 1");
    console.assert(processRequest(1) === 1, "Expected factorial of 1 modulo 10007 is 1");
    console.assert(processRequest(2) === 2, "Expected factorial of 2 modulo 10007 is 2");
    console.assert(processRequest(3) === 6, "Expected factorial of 3 modulo 10007 is 6");
    console.assert(processRequest(4) === 24, "Expected factorial of 4 modulo 10007 is 24");
    console.assert(processRequest(10) === 6266, "Expected factorial of 10 modulo 10007 is 6266");
    console.assert(processRequest(10000) === 6991, "Expected factorial of 10000 modulo 10007 is 6991");

    // console.log("All tests passed.");
}

testProcessRequest();

File: MultiLanguageSandbox/src/test/deps/cpp/6.cpp
#include <iostream>
#include <cassert>
#include <cmath>
#include <cstdio>
#include <cstring>
/**
 * Calculates the factorial of N modulo 10007.
 * 
 * Parameters:
 * - N (int): An integer representing the input value (N <= 10000).
 * 
 * Returns:
 * int: The result after calculating the factorial of N and
 * taking the modulo 10007.
 * 
 * Examples:
 * >>> process_request(1)
 * 1
 * >>> process_request(10)
 * 6266
 */
int process_request(int n) {
    if (n < 0) return 0; // Factorial is not defined for negative numbers
    if (n == 0 || n == 1) return 1; // 0 and 1 are both 1

    int result = 1;
    for (int i = 2; i <= n; ++i) {
        result = (result * i) % 10007; // Multiply and take modulo at each step to prevent overflow
    }
    return result;
}

// Example usage:
int main()
{
    assert(process_request(0) == 1); // Added test for boundary condition
    assert(process_request(1) == 1);
    assert(process_request(2) == 2);
    assert(process_request(3) == 6);
    assert(process_request(4) == 24);
    assert(process_request(10) == 6266); // 10! % 10007 = 3628800 % 10007 = 6266
    assert(process_request(10000) == 6991); // Added a test for upper boundary condition
    // printf("All tests passed.\n");
    return 0;
}